### PR TITLE
add new use case for subtraction, division and multiplication

### PR DESCRIPTION
**Pull request by cesar-ferreira**

### CODE REVIEW

File: app/src/main/java/br/com/cantinho/android_mvvm_clean_architecture/ui/calculator/usecases/AddictionUseCase.kt [hunk 1]
```diff
@@ -3,9 +3,9 @@ package br.com.cantinho.android_mvvm_clean_architecture.ui.calculator.usecases
 import br.com.cantinho.android_mvvm_clean_architecture.ui.calculator.OperationResult
 import br.com.cantinho.android_mvvm_clean_architecture.ui.calculator.Success
 
-class SumUseCase {
+class AddictionUseCase {
 
-    fun performSum(paramA: Int, paramB: Int): OperationResult {
+    fun performAddiction(paramA: Int, paramB: Int): OperationResult {
```
Comment (samirtf) [inline app/src/main/java/br/com/cantinho/android_mvvm_clean_architecture/ui/calculator/usecases/AddictionUseCase.kt]: I think the user case name is already meaningfully. Maybe it's better change method's name `performAddiction` to `perform` only.

File: app/src/main/java/br/com/cantinho/android_mvvm_clean_architecture/ui/calculator/usecases/DivisionUseCase.kt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package br.com.cantinho.android_mvvm_clean_architecture.ui.calculator.usecases
+
+import br.com.cantinho.android_mvvm_clean_architecture.ui.calculator.OperationResult
+import br.com.cantinho.android_mvvm_clean_architecture.ui.calculator.Success
+
+class DivisionUseCase {
+
+    fun performDivision(paramA: Int, paramB: Int): OperationResult {
```
Comment (samirtf) [inline app/src/main/java/br/com/cantinho/android_mvvm_clean_architecture/ui/calculator/usecases/DivisionUseCase.kt]: I think the user case name is already meaningfully. Maybe it's better change method's name `performDivision` to perform only.

File: app/src/main/java/br/com/cantinho/android_mvvm_clean_architecture/ui/calculator/usecases/MultiplicationUseCase.kt [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package br.com.cantinho.android_mvvm_clean_architecture.ui.calculator.usecases
+
+import br.com.cantinho.android_mvvm_clean_architecture.ui.calculator.OperationResult
+import br.com.cantinho.android_mvvm_clean_architecture.ui.calculator.Success
+
+class MultiplicationUseCase {
+
+    fun performMultiplication(paramA: Int, paramB: Int): OperationResult {
```
Comment (samirtf) [inline app/src/main/java/br/com/cantinho/android_mvvm_clean_architecture/ui/calculator/usecases/MultiplicationUseCase.kt]: I think the user case name is already meaningfully. Maybe it's better change method's name `performMultiplication` to perform only.

File: app/src/main/java/br/com/cantinho/android_mvvm_clean_architecture/ui/calculator/usecases/SubtractionUseCase.kt [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package br.com.cantinho.android_mvvm_clean_architecture.ui.calculator.usecases
+
+import br.com.cantinho.android_mvvm_clean_architecture.ui.calculator.OperationResult
+import br.com.cantinho.android_mvvm_clean_architecture.ui.calculator.Success
+
+class SubtractionUseCase {
+
+    fun performSubtraction(paramA: Int, paramB: Int): OperationResult {
```
Comment (samirtf) [inline app/src/main/java/br/com/cantinho/android_mvvm_clean_architecture/ui/calculator/usecases/SubtractionUseCase.kt]: I think the user case name is already meaningfully. Maybe it's better change method's name `performSubtraction` to perform only.

File: app/src/main/res/layout/activity_calculator.xml [hunk 5]
```diff
@@ -33,20 +33,21 @@
...
     </com.google.android.material.textfield.TextInputLayout>
 
     <Button
-        android:id="@+id/btnPerformOperation"
+        android:id="@+id/btnPerformOperationAddiction"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
         android:layout_marginBottom="8dp"
-        android:text="Button"
+        android:text="+"
```
Comment (samirtf) [inline app/src/main/res/layout/activity_calculator.xml]: Maybe doing some internationalization here can be good.

File: app/src/main/res/layout/activity_calculator.xml [hunk 6]
```diff
@@ -58,8 +59,47 @@
...
         app:layout_constraintTop_toBottomOf="@+id/textInputLayout2" />
+
+    <Button
+        android:id="@+id/btnPerformOperationSubtraction"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="8dp"
+        android:layout_marginLeft="8dp"
+        android:layout_marginBottom="8dp"
+        android:text="-"
```
Comment (samirtf) [inline app/src/main/res/layout/activity_calculator.xml]: Maybe doing some internationalization here can be good.

File: app/src/main/res/layout/activity_calculator.xml [hunk 7]
```diff
@@ -58,8 +59,47 @@
...
+        app:layout_constraintStart_toEndOf="@+id/btnPerformOperationAddiction" />
+
+    <Button
+        android:id="@+id/btnPerformOperationMultiplication"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="8dp"
+        android:layout_marginLeft="8dp"
+        android:layout_marginBottom="8dp"
+        android:text="*"
```
Comment (samirtf) [inline app/src/main/res/layout/activity_calculator.xml]: Maybe doing some internationalization here can be good.

File: app/src/main/res/layout/activity_calculator.xml [hunk 8]
```diff
@@ -58,8 +59,47 @@
...
+        app:layout_constraintStart_toEndOf="@+id/btnPerformOperationSubtraction" />
+
+    <Button
+        android:id="@+id/btnPerformOperationDivision"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="8dp"
+        android:layout_marginLeft="8dp"
+        android:layout_marginBottom="8dp"
+        android:text="/"
```
Comment (samirtf) [inline app/src/main/res/layout/activity_calculator.xml]: Maybe doing some internationalization here can be good.

**Review by samirtf — COMMENTED** [review]
Your review is ready.

File: app/src/main/java/br/com/cantinho/android_mvvm_clean_architecture/ui/calculator/usecases/AddictionUseCase.kt (hunk 1 above)
Comment (cesar-ferreira) [reply app/src/main/java/br/com/cantinho/android_mvvm_clean_architecture/ui/calculator/usecases/AddictionUseCase.kt]: resolved

File: app/src/main/java/br/com/cantinho/android_mvvm_clean_architecture/ui/calculator/usecases/MultiplicationUseCase.kt (hunk 3 above)
Comment (cesar-ferreira) [reply app/src/main/java/br/com/cantinho/android_mvvm_clean_architecture/ui/calculator/usecases/MultiplicationUseCase.kt]: resolved

File: app/src/main/res/layout/activity_calculator.xml (hunk 5 above)
Comment (cesar-ferreira) [reply app/src/main/res/layout/activity_calculator.xml]: resolved

File: app/src/main/res/layout/activity_calculator.xml (hunk 6 above)
Comment (cesar-ferreira) [reply app/src/main/res/layout/activity_calculator.xml]: would it be necessary to internationalize buttons with arithmetic operations?

File: app/src/main/res/layout/activity_calculator.xml (hunk 7 above)
Comment (cesar-ferreira) [reply app/src/main/res/layout/activity_calculator.xml]: would it be necessary to internationalize buttons with arithmetic operations?

File: app/src/main/res/layout/activity_calculator.xml (hunk 8 above)
Comment (cesar-ferreira) [reply app/src/main/res/layout/activity_calculator.xml]: would it be necessary to internationalize buttons with arithmetic operations?

File: app/src/main/res/layout/activity_calculator.xml (hunk 5 above)
Comment (cesar-ferreira) [reply app/src/main/res/layout/activity_calculator.xml]: would it be necessary to internationalize buttons with arithmetic operations?

Comment (cesar-ferreira) [reply app/src/main/res/layout/activity_calculator.xml]: resolved

File: app/src/main/res/layout/activity_calculator.xml (hunk 6 above)
Comment (cesar-ferreira) [reply app/src/main/res/layout/activity_calculator.xml]: resolved

File: app/src/main/res/layout/activity_calculator.xml (hunk 7 above)
Comment (cesar-ferreira) [reply app/src/main/res/layout/activity_calculator.xml]: resolved

File: app/src/main/res/layout/activity_calculator.xml (hunk 8 above)
Comment (cesar-ferreira) [reply app/src/main/res/layout/activity_calculator.xml]: resolved

**Comment by samirtf** [comment]
Reviewed by @samirtf .

**Pull request merged**